v7.0.0-beta.2 (Holesky Rescue)

paulhauner opened the pull request:
## Issue Addressed

Related to https://github.com/sigp/lighthouse/issues/7040

## Proposed Changes

Various hacks and tweaks to ensure LH syncs a valid Holesky chain.

Most notably, we explicitly reject the invalid payload that was justified (`0x2db899881ed8546476d0b92c6aa9110bea9a4cd0dbeb5519eb0ea69575f1f359`).


## Additional Info

- **Use this branch if you are running a node on Holesky**

michaelsproul committed:
Add flag to disable attestation APIs

michaelsproul committed:
Fix flag

michaelsproul committed:
Disable liveness risk

michaelsproul committed:
lcli http-sync hacks

michaelsproul committed:
Remove more liveness risks

michaelsproul committed:
Implement invalidation API

michaelsproul committed:
Allow invalidation of "valid" nodes

michaelsproul committed:
MORE

jimmygchen committed:
Blacklist invalid block root in block verification and blacklist invalid finalized epochs in sync.

jimmygchen committed:
Blacklist invalid block root in block verification and blacklist invalid finalized epochs in sync.

jimmygchen committed:
Add log

michaelsproul committed:
Ban peers with banned payloads

michaelsproul committed:
Add vc --disable-attesting flag

pawanjay176 committed:
Bump sync-tolerance-epoch and make it a cli param

pawanjay176 committed:
take config value

eserilev committed:
discard unused code

paulhauner committed:
Bump version

michaelsproul commented on validator_client/src/lib.rs:
```diff
@@ -325,7 +325,7 @@ impl<E: EthSpec> ProductionValidatorClient<E> {
                 }
             } else {
-                Timeouts::set_all(slot_duration)
+                Timeouts::set_all(25 * slot_duration)
```
@macladson could you make a clean PR for a CLI flag accomplishing the same effect against the `release-v7.0.0` branch? :pray:

michaelsproul replied:
PR is now here: https://github.com/sigp/lighthouse/pull/7047

jimmygchen committed:
Optimise status processing for holesky-rescue (#7054)

* Optimise status processing

* Fix lint

---------

Co-authored-by: Michael Sproul <michael@sigmaprime.io>

jimmygchen committed:
Load block roots from fork choice where possible when serving `BlocksByRange` requests (#7058)

* Load block roots from fork choice where possible to avoid loading state from disk when serving block by range requests.

* Check if the start slot is newer than finalization (`start_slot >= finalized_slot`), and use fork choice in that case.

jimmygchen committed:
Change state cache size default to 32. (#7055)

eserilev committed:
Manual finalization endpoint (#7059)

* Load block roots from fork choice where possible to avoid loading state from disk when serving block by range requests.

* Check if the start slot is newer than finalization (`start_slot >= finalized_slot`), and use fork choice in that case.

* force finalization endpoint

* cleanup

* Remove ds store

* Don't import blocks that conflict with the split

* Disconnect and ban peer if we get blocks conflicting manual checkpoint

* immediately commit state to cold db

* revert

* Fix descent from split check

* Add safety check to checkpoint when doing manual finalization.

---------

Co-authored-by: Jimmy Chen <jchen.tc@gmail.com>
Co-authored-by: Michael Sproul <michael@sigmaprime.io>
Co-authored-by: Pawan Dhananjay <pawandhananjay@gmail.com>

michaelsproul committed:
Reuse milhouse subtrees to shrink inactivity_scores in memory (#7062)

* SSZ bump in a single clean commit

* Do the actual intra-rebases

* Update milhouse

michaelsproul committed:
Revert "Reuse milhouse subtrees to shrink inactivity_scores in memory (#7062)"

This reverts commit 6399ad482ba500c18d81ea5d0bf11cc87419718c.

realbigsean committed:
Split block root lookups between fork choice and store on BBR response (#7066)

* handle bbr across finalized slot gracefully

* sort deps

* Michael's off-by-one corrections

---------

Co-authored-by: Michael Sproul <micsproul@gmail.com>

michaelsproul commented:
Trying to get CI running/passing on this so we have a bit more confidence in the release.

eserilev committed:
Prevent writing to state cache when migrating the database (#7067)

* add an update_cache flag to get_state to have more granular control over when we write to the cache

* State cache tweaks

- add state-cache-headroom flag to control pruning
- prune old epoch boundary states ahead of mid-epoch states
- never prune head block's state
- avoid caching ancestor states unless they are on an epoch boundary

---------

Co-authored-by: Michael Sproul <michael@sigmaprime.io>

jimmygchen committed:
Add backtrace logging. (#7063)

michaelsproul commented on beacon_node/store/src/hot_cold_store.rs:
```diff
@@ -1505,34 +1518,50 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
-                "state_root" => ?state_root,
-                "slot" => state.slot(),
-            );
+            if update_cache {
+                state.update_tree_hash_cache()?;
+                state.build_all_caches(&self.spec)?;
```
@jimmygchen This is why the test is failing, we are now only building the state caches when `update_cache` is `true`. We should do it all the time so that the caller gets nice states with caches built.

jimmygchen replied:
Nice, I've made a fix in https://github.com/sigp/lighthouse/pull/7071. Thanks!

pawanjay176 committed:
Add http endpoint to add trusted peer (#7068)

* Add a trusted_peers endpoint

* Maintain trusted peers in heartbeat

jimmygchen committed:
Add CI fixes to `holesky-rescue` (#7071)

* Add CI fixes.

* Remove log and update cli docs.

* Build caches after loading state from disk.

* Revert payload invalidation API - wasn't working because of the justification being permanently updated.

eserilev committed:
Manual compaction endpoint (#7072)

* manual compaction endpoint

* linting

---------

Co-authored-by: Michael Sproul <michael@sigmaprime.io>

jimmygchen committed:
Bump version to v7.0.0-beta.2 (#7073)

michaelsproul committed:
Fix descent from split check (#7105)

michaelsproul committed:
Release v7.0.0-beta.3

michaelsproul commented:
Closing as no longer relevant.
